azkaban-common/src/main/java/azkaban/executor/container/ContainerizedExecutionManager.java

```diff
@@ -0,0 +1,383 @@
+      this.azkProps = azkProps;
+      this.executorLoader = executorLoader;
+      setActive(
+          this.azkProps.getBoolean(Constants.ConfigurationKeys.QUEUEPROCESSING_ENABLED, true));
+      this.queueProcessorWaitInMS =
+          this.azkProps.getLong(Constants.ConfigurationKeys.QUEUE_PROCESSOR_WAIT_IN_MS, 1000);
+      this.executionsBatchProcessingEnabled = azkProps
+          .getBoolean(ContainerizedExecutionManagerProperties.CONTAINERIZED_EXECUTION_BATCH_ENABLED,
+              false);
+      this.executionsBatchSize =
+          azkProps
+              .getInt(ContainerizedExecutionManagerProperties.CONTAINERIZED_EXECUTION_BATCH_SIZE,
+                  10);
+
+      this.executorService = Executors.newFixedThreadPool(azkProps.getInt(
+          ContainerizedExecutionManagerProperties.CONTAINERIZED_EXECUTION_PROCESSING_THREAD_SIZE,
+          10));
+      this.setName("Containerized-QueueProcessor-Thread");
+    }
+
+    @Override
+    public void run() {
+      // Loops till QueueProcessorThread is shutdown
+      while (!this.shutdown) {
+        synchronized (this) {
+          try {
+            // start processing queue if active, otherwise wait for sometime
+            if (this.isActive) {
+              processQueuedFlows();
+            }
+            wait(queueProcessorWaitInMS);
+          } catch (final Exception e) {
+            ContainerizedExecutionManager.logger.error(
+                "QueueProcessorThread Interrupted. Probably to shut down.", e);
+          }
+        }
+      }
+    }
+
+    /* Method responsible for processing the non-dispatched flows */
+    private void processQueuedFlows() throws ExecutorManagerException {
+      Set<Integer> executionIds =
+          executorLoader.selectAndUpdateExecutionWithLocking(this.executionsBatchProcessingEnabled, this.executionsBatchSize,
+              Status.DISPATCHING);
+
+      for (int executionId : executionIds) {
+        Runnable worker = new ExecutionDispatcher(executionId);
+        executorService.execute(worker);
+      }
+    }
+
+    public boolean isActive() {
+      return this.isActive;
+    }
+
+    public void setActive(final boolean isActive) {
+      this.isActive = isActive;
+      ContainerizedExecutionManager.logger
+          .info("QueueProcessorThread active turned " + this.isActive);
+    }
+
+    public void shutdown() {
+      this.shutdown = true;
+      this.executorService.shutdown();
+      this.interrupt();
+    }
+  }
+
+  /**
+   * This class is a worker class to dispatch execution in container.
+   */
+  private class ExecutionDispatcher implements Runnable {
+
+    private final int executionId;
+
+    ExecutionDispatcher(final int executionId) {
+      this.executionId = executionId;
+    }
+
+    @Override
+    public void run() {
+      try {
+        ContainerizedExecutionManager.this.containerizedImpl.createContainer(executionId);
```

Comment:
Executing the k8s API for Deployment/Stateful set creation etc will typically not return an error immediately (may do some basic yaml parsing). 
Is the `createContainer` call expected to perform some additional checks beyond creating the deployment and service resources ?
(would be good to mention it in comments either way)

Reply:
There are a lot of exception that could come like templates not available, version set API issue, parsing error. I will add all of these things in comment.